language-plutus-core/test/Evaluation/Machines/Budget/Fib/1.plc.golden

```diff
@@ -5,8 +5,8 @@
 | BApply causes ({ cpu: 18
 | mem: 18
 })
-| BVar causes ({ cpu: 14
-| mem: 14
+| BVar causes ({ cpu: 13
```

Comment:
A bit odd that this changes, do we know why?

Reply:
That's probably because in the previous version builtins could only handle constants and variables, and if we wanted to pass a term to a builtin we had to bind it to a fresh variable first (see `withScopedArgIn` in the CEK machine in master).  Roman generalised the constant application machinery so that it now accepts an opaque type of `term`, and we don''t need generate fresh variables any more.  Having said that, `master` contains the new machinery but still uses `withScopedArgIn`; maybe @effectfully just didn't update that because he was expecting this to be merged.

Reply:
> Roman generalised the constant application machinery so that it now accepts an opaque type of `term`, and we don''t need generate fresh variables any more. 

No, that's not the case. We need an opaque type of `term` in order to pass anything through CAM (`CekVal`, untyped terms, whatever). That does not solve the problem of needing to generate a fresh variable in order to store an environment of a term _somewhere_, which was the problem with the original CEK machine. The alternative CEK machine does not have this problem, because environments are stored inside values and so we don't need any such trickery and correct evaluation of polymorphic built-in functions is supported out of the box.

Reply:
OK, sorry. I think that still explains why the resource consumption will have changed though.